sdk/trace/batch_span_processor_test.go

```diff
@@ -474,6 +493,38 @@ func TestBatchSpanProcessorForceFlushSucceeds(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestBatchSpanProcessorLogsWarningOnNewDropSpans(t *testing.T) {
+	tLog := testr.NewWithOptions(t, testr.Options{Verbosity: 1})
+	l := &logCounter{LogSink: tLog.GetSink()}
+	otel.SetLogger(logr.New(l))
+	global.SetLogger(logr.New(l))
+
+	te := testBatchExporter{}
+	tp := basicTracerProvider(t)
+	option := testOption{
+		name: "default BatchSpanProcessorOptions",
+		o: []sdktrace.BatchSpanProcessorOption{
+			sdktrace.WithMaxQueueSize(0),
+			sdktrace.WithMaxExportBatchSize(10),
+		},
+		genNumSpans: 100,
+	}
+	ssp := createAndRegisterBatchSPNonBlocking(option, &te)
+	if ssp == nil {
+		t.Fatalf("%s: Error creating new instance of BatchSpanProcessor\n", option.name)
+	}
+	tp.RegisterSpanProcessor(ssp)
+	tr := tp.Tracer("BatchSpanProcessorWithOption")
+
+	// Force flush any held span batches
+	generateSpan(t, tr, option)
+	ssp.ForceFlush(context.Background())
+
+	assert.Equal(t, len(l.logs), 3, "expected log message")
+	droppedLogs := l.logs[2]
```

Comment:
This seems brittle, if additional logging is added or removed this will break. Instead, could the entire slice be asserted to contain at least one conforming log message?

Reply:
Yes I can revisit this, do you have any suggestions about the race test failing when using the non-blocking code path?  
(described here in one of previous comments above)
https://github.com/open-telemetry/opentelemetry-go/pull/3289#issuecomment-1282971005

Reply:
Hmm, nothing off hand. I would need to think about it.

Reply:
I think i got it passing, it seemed to be an issue with setting global logger in the test & not resetting it.  I've ran `make test-race` a dozen or so times and it seems consistent now.